Modules/mathmodule.c

```diff
@@ -2031,49 +2031,64 @@ math_fmod_impl(PyObject *module, double x, double y)
         return PyFloat_FromDouble(r);
 }
 
-
-/*[clinic input]
-math.hypot
-
-    x: double
-    y: double
-    /
-
-Return the Euclidean distance, sqrt(x*x + y*y).
-[clinic start generated code]*/
-
+/* AC: cannot convert yet, waiting for *args support */
 static PyObject *
-math_hypot_impl(PyObject *module, double x, double y)
-/*[clinic end generated code: output=b7686e5be468ef87 input=7f8eea70406474aa]*/
+math_hypot(PyObject *self, PyObject *args)
 {
-    double r;
-    /* hypot(x, +/-Inf) returns Inf, even if x is a NaN. */
-    if (Py_IS_INFINITY(x))
-        return PyFloat_FromDouble(fabs(x));
-    if (Py_IS_INFINITY(y))
-        return PyFloat_FromDouble(fabs(y));
-    errno = 0;
-    PyFPE_START_PROTECT("in math_hypot", return 0);
-    r = hypot(x, y);
-    PyFPE_END_PROTECT(r);
-    if (Py_IS_NAN(r)) {
-        if (!Py_IS_NAN(x) && !Py_IS_NAN(y))
-            errno = EDOM;
-        else
-            errno = 0;
+    Py_ssize_t i, n;
+    PyObject *item;
+    double max = 0.0;
+    double csum = 0.0;
+    double x;
+    int found_inf = 0;
+    int found_nan = 0;
+
+    n = PyTuple_GET_SIZE(args);
+    for (i=0 ; i<n ; i++) {
+        item = PyTuple_GET_ITEM(args, i);
+        x = PyFloat_AsDouble(item);
+        if (x == -1.0 && PyErr_Occurred()) {
+            return NULL;
+        }
+        found_inf |= Py_IS_INFINITY(x);
+        found_nan |= Py_IS_NAN(x);
+        x = fabs(x);
+        if (x > max) {
+            max = x;
+        }
     }
-    else if (Py_IS_INFINITY(r)) {
-        if (Py_IS_FINITE(x) && Py_IS_FINITE(y))
-            errno = ERANGE;
-        else
-            errno = 0;
+    if (found_inf) {
+        return PyFloat_FromDouble(max);
     }
-    if (errno && is_error(r))
-        return NULL;
-    else
-        return PyFloat_FromDouble(r);
+    if (found_nan) {
+        return PyFloat_FromDouble(Py_NAN);
+    }
+    if (max == 0.0) {
+        return PyFloat_FromDouble(0.0);
+    }
+    for (i=0 ; i<n ; i++) {
+        item = PyTuple_GET_ITEM(args, i);
+        x = PyFloat_AsDouble(item) / max;
+        csum += x * x;
+    }
+    return PyFloat_FromDouble(max * sqrt(csum));       // XXX Handle overflow
```

Comment:
It's good enough for me that +Inf will result from overflow here, but that it is a change from, e.g., what 3.7 does on Windows:

    >>> import sys
    >>> import math
    >>> x = sys.float_info.max
    >>> math.hypot(x, x)
    Traceback (most recent call last):
        ...
    OverflowError: math range error

Reply:
That is good enough for me too :-)   I like that it matches the behavior of the pure python code: `scale * sqrt(sum((x/scale) ** x for x in coordinates))`

    >>> import sys
    >>> coordinates = [sys.float_info.max] * 1
    >>> scale * sqrt(sum((x/scale) ** x for x in coordinates))
    1.7976931348623157e+308
    >>> coordinates = [sys.float_info.max] * 2
    >>> scale * sqrt(sum((x/scale) ** x for x in coordinates))
    inf